common/src/main/java/com/wynntils/screens/WynntilsQuestBookScreen.java

```diff
@@ -0,0 +1,513 @@
+                true,
+                this));
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        renderBackgroundTexture(poseStack);
+
+        // Make 0, 0 the top left corner of the rendered quest book background
+        poseStack.pushPose();
+        final float translationX = getTranslationX();
+        final float translationY = getTranslationY();
+        poseStack.translate(translationX, translationY, 1f);
+
+        renderTitle(poseStack);
+
+        renderVersion(poseStack);
+
+        renderButtons(poseStack, mouseX, mouseY, partialTick);
+
+        if (quests.isEmpty()) {
+            renderNoQuestsHelper(poseStack);
+        }
+
+        renderDescription(poseStack);
+
+        renderPageInfo(poseStack);
+
+        poseStack.popPose();
+
+        renderTooltip(poseStack, mouseX, mouseY);
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        final float translationX = getTranslationX();
+        final float translationY = getTranslationY();
+
+        for (GuiEventListener child : new ArrayList<>(this.children())) {
+            if (child.isMouseOver(mouseX - translationX, mouseY - translationY)) {
+                child.mouseClicked(mouseX - translationX, mouseY - translationY, button);
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+        if (keyCode == GLFW.GLFW_KEY_ESCAPE && this.shouldCloseOnEsc()) {
+            McUtils.mc().setScreen(null);
+            return true;
+        }
+
+        if (searchWidget != null) {
+            return searchWidget.keyPressed(keyCode, scanCode, modifiers);
+        }
+
+        return super.keyPressed(keyCode, scanCode, modifiers);
+    }
+
+    @Override
+    public boolean charTyped(char codePoint, int modifiers) {
+        if (searchWidget != null) {
+            return searchWidget.charTyped(codePoint, modifiers);
+        }
+
+        return super.charTyped(codePoint, modifiers);
+    }
+
+    @Override
+    public boolean mouseScrolled(double mouseX, double mouseY, double delta) {
+        if (delta < 0) {
+            setCurrentPage(getCurrentPage() + 1);
+        } else if (delta > 0) {
+            setCurrentPage(getCurrentPage() - 1);
+        }
+
+        return true;
+    }
+
+    @SubscribeEvent
+    public void onQuestsReloaded(QuestBookReloadedEvent event) {
+        this.setQuests(QuestBookManager.getQuests());
+
+        for (QuestInfo quest : quests) {
+            if (!quest.isTracked()) {
+                continue;
+            }
+
+            tracked = quest;
+            return;
+        }
+    }
+
+    // FIXME: We only need this hack to stop the screen from closing when tracking Quest.
+    //        Adding a proper way to add quests with scripted container queries would mean this can get removed.
```

Comment:
After reading through all the code, I don't get this. You are tracking using scripted container queries..? Is this from an older iteration?

Reply:
I use a modified version of quest analysis query for tracking a quest. But clicking on a quest to track force-closes the container, "breaking" the query. In reality, it sends a warning, but works.

Reply:
Hm, I might need to help you craft the query, since that should work. You just have to make sure you tell the query maanger that you expect the last click to close the container. I explicitly designed it to handle this very use case, even though I never quite got around to doing the `/quest track` that I was planning. (see https://github.com/magicus/Artemis/tree/quest-book-stuff)

Reply:
Basically, my thinking was that I should get all the QuestBookManager stuff in place using commands, so when writing the GUI we'd only need to consider the UI stuff. I never got further than this when I ran out of time to spend on Wynntils, though. :-(

https://github.com/Wynntils/Artemis/compare/main...magicus:Artemis:quest-book-stuff?expand=1

Reply:
Yeah, I would appreciate if you could help me with this part.

Reply:
I’ll see what I can do. If I do not have time to do anything right now, push this as it is and I’ll rewrite it afterwards and remove the workarounds.